### PR TITLE
Enabled CV ops with plain layout

### PR DESCRIPTION
**Pull request by crazydemo**
Enabled OP List:

- Convolution 1D-3D
- Deconvolution 2D-3D
- Pooling 2D-3D (avg / max)
- Elementwise OPs (abs, clip, exp, log, sqrt, round, logsumexp, relu, leaky_relu, tanh, sigmoid, softmax)

### CODE REVIEW

File: src/relay/backend/contrib/dnnl/codegen.cc [hunk 1]
```diff
@@ -473,9 +493,87 @@ class DNNLJSONSerializer : public backend::contrib::JSONSerializer {
...
+        ICHECK(call->op.as<OpNode>()) << "Not op node";
+      } else if (name == "dnnl.dense_relu") {
+        call = GetRootCall(fn->body.as<CallNode>(), 1, {"nn.dense", "nn.relu"});
+        ICHECK(call->op.as<OpNode>()) << "Not op node";
+      } else if (name == "dnnl.dense_tanh") {
+        call = GetRootCall(fn->body.as<CallNode>(), 1, {"nn.dense", "tanh"});
+        ICHECK(call->op.as<OpNode>()) << "Not op node";
+      } else if (name == "dnnl.dense_sigmoid") {
+        call = GetRootCall(fn->body.as<CallNode>(), 1, {"nn.dense", "sigmoid"});
+        ICHECK(call->op.as<OpNode>()) << "Not op node";
```
Comment (ciyongch) [inline src/relay/backend/contrib/dnnl/codegen.cc:497]: We might split these `if-then-else` clauses into several section for different ops with post ops like `dense/conv`, etc, with different post -op in their own section to improve the readability of the codes?

File: src/runtime/contrib/dnnl/dnnl_json_runtime.cc [hunk 2]
```diff
@@ -90,44 +91,95 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
...
+      {"GIODHW", tag::giodhw},
+  };
+
+  std::map<std::string, dnnl::algorithm> elt_name2algo{
+      {"abs", dnnl::algorithm::eltwise_abs},
+      {"exp", dnnl::algorithm::eltwise_exp},
+      {"log", dnnl::algorithm::eltwise_log},
+      {"sqrt", dnnl::algorithm::eltwise_sqrt},
+      {"round", dnnl::algorithm::eltwise_round},
+      {"logsumexp", dnnl::algorithm::eltwise_logsigmoid},
```
Comment (ciyongch) [inline src/runtime/contrib/dnnl/dnnl_json_runtime.cc:107]: Is this mapping exactly the same?

File: src/runtime/contrib/dnnl/dnnl_json_runtime.cc [hunk 3]
```diff
@@ -166,100 +218,88 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
...
+    }
 
     // Memory shapes.
-    dnnl::memory::dims src_dims = {N, IC, IH, IW};
-    dnnl::memory::dims weights_dims = {OC, IC, KH, KW};
+    dnnl::memory::dims src_dims = input_shape;       // {N, IC, ID, IH, IW}
+    dnnl::memory::dims weights_dims = weight_shape;  // {OC, IC, KD, KH, KW}
     if (groups > 1) {
-      weights_dims = {groups, 1, IC / groups, KH, KW};
+      weights_dims = {groups, 1, input_shape[1] / groups};
```
Comment (ciyongch) [inline src/runtime/contrib/dnnl/dnnl_json_runtime.cc]: What's the value for `oc` dimension?

File: src/runtime/contrib/dnnl/dnnl_json_runtime.cc [hunk 4]
```diff
@@ -269,24 +309,126 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
...
+    // Weight memory.
+    auto deconv2d_weights_memory = BindDNNLMemory(weight_entry, deconv_weights_md);
+
+    // Bias memory.
+    auto deconv2d_bias_memory = dnnl::memory({bias_dims, dt::f32, tag::x}, engine_);
+    if (has_bias) {
+      auto bias_entry = node.GetInputs()[2];
+      BindDNNLMemory(bias_entry, deconv2d_bias_memory);
+    } else {
+      float bias[channels] = {0};
```
Comment (ciyongch) [inline src/runtime/contrib/dnnl/dnnl_json_runtime.cc]: no need to create and push this arg when `has_bias=false`?

File: src/relay/backend/contrib/dnnl/codegen.cc (hunk 1 above)
Comment (crazydemo) [reply src/relay/backend/contrib/dnnl/codegen.cc:497]: fixed. split the clauses.

File: src/runtime/contrib/dnnl/dnnl_json_runtime.cc (hunk 2 above)
Comment (crazydemo) [reply src/runtime/contrib/dnnl/dnnl_json_runtime.cc:107]: Not exactly. `logsumexp` is supposed to be mapped to `logsigmoid`.

File: src/runtime/contrib/dnnl/dnnl_json_runtime.cc (hunk 3 above)
Comment (crazydemo) [reply src/runtime/contrib/dnnl/dnnl_json_runtime.cc]: replace `1` with `channels / groups`, which equals to `OC`.

File: src/runtime/contrib/dnnl/dnnl_json_runtime.cc (hunk 4 above)
Comment (crazydemo) [reply src/runtime/contrib/dnnl/dnnl_json_runtime.cc]: no need to create and push. fixed.

**Review by ciyongch — APPROVED** [review]

**Pull request closed without merging**